### ISSUE TITLE
Cloudwatch sink is not working after promoting our vector instances into v0.36.0

### ISSUE THREAD

**Issue by rfum** [vectordotdev/vector#19951]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

We were on `v0.35.1` before switching into `v0.36.0`. Everything were working well before switching to the new version. When we switched to the new version, we've ended up with the following log messages on vector instances. At the same time, vector stopped sending new logs into cloud watch sink and all of the vector instances are working with 100% cpu utilization. Vector is deployed as a daemonset in our setup and all the pods are affected with this problem. In order to overcome this issue we have reverted back into previous version and everything works as just like before the version update.I've checked IAM policies and it seems , this is not related to IAM policies because the aws api call that is mentioned as failing in the logs is already approved by our IAM role policy that is attached to all vector instances that we have. Besides, since only reverting back to previous version solves the problem I don't think it's related to our configuration. That might be a breaking problem for many production deployments. Is that a known issue?
```
2024-02-26T10:29:44.359923Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:29:44.593611Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:29:46.500933Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:29:48.623451Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:29:55.716507Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:29:55.718945Z  WARN sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::util::retries: Internal log [Retrying after error.] has been suppressed 5 times.
2024-02-26T10:29:55.718958Z  WARN sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::util::retries: Retrying after error. error=CloudwatchError::DescribeLogStreams: dispatch failure internal_log_rate_limit=true
2024-02-26T10:30:05.911070Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:30:05.913458Z  WARN sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::util::retries: Retrying after error. error=CloudwatchError::DescribeLogStreams: dispatch failure internal_log_rate_limit=true
2024-02-26T10:30:11.247722Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:30:11.249997Z  WARN sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::util::retries: Internal log [Retrying after error.] is being suppressed to avoid flooding.
2024-02-26T10:30:38.575236Z  INFO sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::aws_cloudwatch_logs::service: Sending events. events=13
2024-02-26T10:30:38.577434Z  WARN sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::util::retries: Internal log [Retrying after error.] has been suppressed 1 times.
2024-02-26T10:30:38.577446Z  WARN sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::util::retries: Retrying after error. error=CloudwatchError::DescribeLogStreams: dispatch failure internal_log_rate_limit=true
2024-02-26T10:30:42.837415Z  WARN sink{component_kind="sink" component_id=cloud_watch component_type=aws_cloudwatch_logs}:request{request_id=1}: vector::sinks::util::retries: Request timed out. If this happens often while the events are actually reaching their destination, try decreasing `batch.max_bytes` and/or using `compression` if applicable. Alternatively `request.timeout_secs` can be increased. internal_log_rate_limit=true
```

### Version

v0.36.0

**Comment by StephenWakely** [comment]
Can you share your config for the `aws_cloudwatch_logs` sink please.

**Comment by StephenWakely** [comment]
Thank you. 

This should be fixed by #19939. I'm just testing that PR now.

**Comment by jszwedko** [comment]
Closed by https://github.com/vectordotdev/vector/pull/19939
